contrib/mpi-proxy-split/split_process.cpp

```diff
@@ -333,7 +333,8 @@ startProxy()
       // Read from stdout of lh_proxy full lh_info struct, including orig memRange.
       close(pipefd_out[1]); // close write end of pipe
       // FIXME: should be a readall. Check for return error code.
-      if (read(pipefd_out[0], &lh_info, sizeof lh_info) < sizeof lh_info) {
+      if ((unsigned long) read(pipefd_out[0], &lh_info, sizeof lh_info)
+        < sizeof lh_info) {
```

Comment:
`read` can return `-1`, make it `unsigned long` will be 2^32 - 1 and passed this condition check, which is not correct.